internal/provider/resource_gitlab_project.go

```diff
@@ -387,11 +403,201 @@ var resourceGitLabProjectSchema = map[string]*schema.Schema{
 		Optional:    true,
 		Default:     false,
 	},
+	"resolve_outdated_diff_discussions": {
+		Description: "Automatically resolve merge request diffs discussions on lines changed with a push.",
+		Type:        schema.TypeBool,
+		Optional:    true,
+	},
+	"analytics_access_level": {
+		Description:      fmt.Sprintf("Set the analytics access level. Valid values are %s.", renderValueListForDocs(validProjectAccessLevels)),
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validProjectAccessLevels, false)),
+	},
+	"auto_cancel_pending_pipelines": {
+		Description:      "Auto-cancel pending pipelines. This isn’t a boolean, but enabled/disabled.",
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validProjectAutoCancelPendingPipelinesValues, false)),
+	},
+	"auto_devops_deploy_strategy": {
+		Description:      fmt.Sprintf("Auto Deploy strategy. Valid values are %s.", renderValueListForDocs(validProjectAutoDevOpsDeployStrategyValues)),
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validProjectAutoDevOpsDeployStrategyValues, false)),
+	},
+	"auto_devops_enabled": {
+		Description: "Enable Auto DevOps for this project.",
+		Type:        schema.TypeBool,
+		Optional:    true,
+	},
+	"autoclose_referenced_issues": {
+		Description: "Set whether auto-closing referenced issues on default branch.",
+		Type:        schema.TypeBool,
+		Optional:    true,
+	},
+	"build_git_strategy": {
+		Description: "The Git strategy. Defaults to fetch.",
```

Comment:
This could use a `ValidateDiagFunc` for values `clone, fetch`. https://gitlab.com/gitlab-org/gitlab/-/blob/89d18cb96aad3e6522228948f6e17bf7f89a9904/lib/api/helpers/projects_helpers.rb#L13